project/UI/Jekyll/Jekyll.swift

```diff
@@ -2,51 +2,50 @@
-    case project
-    case output
-    case main = "main-page"
-}
-
+public struct JekyllCommand: ConsoleCommand {
+    public static var commandName: String = "nef-jekyll"
+    public static var configuration = CommandConfiguration(commandName: commandName,
+                                                           abstract: "Render Markdown files that can be consumed from Jekyll to generate a microsite")
 
-@discardableResult
-public func jekyll(script: String) -> Either<CLIKit.Console.Error, Void> {
+    public init() {}
     
-    func arguments(console: CLIKit.Console) -> IO<CLIKit.Console.Error, (input: URL, output: URL, mainPage: URL)> {
-        console.input().flatMap { args in
-            guard let inputPath = args[JekyllCommand.project.rawValue]?.trimmingEmptyCharacters.expandingTildeInPath,
-                  let outputPath = args[JekyllCommand.output.rawValue]?.trimmingEmptyCharacters.expandingTildeInPath,
-                  let mainPagePath = args[JekyllCommand.main.rawValue]?.trimmingEmptyCharacters.expandingTildeInPath else {
-                    return IO.raiseError(.arguments)
-            }
-            
-            let input = URL(fileURLWithPath: inputPath, isDirectory: true)
-            let output = URL(fileURLWithPath: outputPath, isDirectory: true)
-            let mainPage = mainPagePath == "README.md" ? output.appendingPathComponent("README.md") : URL(fileURLWithPath: mainPagePath, isDirectory: false)
-            
-            return IO.pure((input: input, output: output, mainPage: mainPage))
-        }^
-    }
+    @ArgumentParser.Option(help: "Path to the Xcode Playground to render")
+    var project: String
+    
+    @ArgumentParser.Option(help: "Path where the resulting jekyll files will be generated")
+    var output: String
+    
+    @ArgumentParser.Option(name: .customLong("main-page"), default: "README.md", help: "Path to 'README.md' file to be used as the index page")
+    var mainPage: String
     
-    let console = Console(script: script,
-                          description: "Render markdown files that can be consumed from Jekyll to generate a microsite",
-                          arguments: .init(name: JekyllCommand.project.rawValue, placeholder: "path-to-input", description: "path to the nef Playground to render"),
-                                     .init(name: JekyllCommand.output.rawValue, placeholder: "path-to-output", description: "path where the resulting Markdown files will be generated"),
-                                     .init(name: JekyllCommand.main.rawValue, placeholder: "path-to-index", description: "path to 'README.md' file to be used as the index page", default: "README.md"))
+    var projectURL: URL { URL(fileURLWithPath: project.trimmingEmptyCharacters.expandingTildeInPath) }
+    var outputURL: URL  { URL(fileURLWithPath: output.trimmingEmptyCharacters.expandingTildeInPath) }
+    var mainPath: String { mainPage.trimmingEmptyCharacters.expandingTildeInPath }
```

Comment:
It seems this pattern `trimmingEmptyCharacters.expandingTildeInPath` is happening recurrently, so, in order to avoid errors, it should be extracted to a computed property, or even to its own type to make sure all Strings that need to be managed like that actually behave like expected.

Consider creating a new type and implementing the `ExpressibleByStringLiteral` protocol.